src/Pages/Channels.razor

```diff
@@ -77,10 +77,17 @@
                 <DataGridColumn TItem="Channel" Field="@nameof(Channel.UpdateDatetime)" Caption="@nameof(Channel.UpdateDatetime).Humanize(LetterCasing.Sentence)" Sortable="false"/>
                 <DataGridColumn TItem="Channel" Caption="Channel Balance">
                     <DisplayTemplate>
-                        <Progress Color="Color.Primary" Value="@Convert.ToInt32(await GetPercentageBalance(context))"/>
-                        <Text>@{
-                                  @($"{Math.Round(await GetPercentageBalance(context), 2)}%")
-                        }</Text>
+                        @{
+                            var balance = Task.Run(()=> GetPercentageBalance(context)).Result;
+                            var percentageBalance = Convert.ToInt32(balance);
+                            var percentageBalanceRound = Math.Round(balance, 2);
+                        }
+                        <Progress Color="Color.Primary" Value="@percentageBalance"/>
+                        <Text>
+                            @{
+                                @($"{percentageBalanceRound}%")
```

Comment:
That approach looks better than mine. But whats the difference of doing it like that and doing an await?

Reply:
@markettes Sometimes you cannot do async on a Razor code, so you need to use Task.Run() (never use .Result directly, it can lead to deadlocks or exceptions not handled).